packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -398,10 +398,10 @@ public void makeDirectoryAsync(String uriStr, Map<String, Object> options, Promi
       ensurePermission(uri, Permission.WRITE);
       if ("file".equals(uri.getScheme())) {
         File file = uriToFile(uri);
-        boolean success = options.containsKey("intermediates") && (Boolean) options.get("intermediates") ?
-                file.mkdirs() :
-                file.mkdir();
-        if (success) {
+        boolean previouslyCreated = file.isDirectory();
+        boolean setIntermediates = options.containsKey("intermediates") && (Boolean) options.get("intermediates");
+        boolean success = setIntermediates ? file.mkdirs() : file.mkdir();
+        if (success || (setIntermediates && previouslyCreated)) {
           promise.resolve(null);
         } else {
           promise.reject("E_DIRECTORY_NOT_CREATED",
```

Comment:
Update the error message so that it reads `...could not be created or already exists`